### PR TITLE
integration of android ndk (for travis build)

### PR DESCRIPTION
**Pull request by arris69**
~~don't merge.~~ finished

some smart tips how to merge it into the buildmatrix?

### CODE REVIEW

**Comment by acgessler** [comment]
Very nice. As for integrating with the build matrix, I'm not aware of any better way than to branch in the `script` section.

**Comment by jdduke** [comment]
Interesting! One concern I saw voiced by developers looking at similar options was the install/build time to setup the Android/NDK environment properly with travis. Build time turnaround may not be a bottleneck now, but it's something to keep in mind.

**Comment by jdduke** [comment]
Looking at the travis result, it appears that the total build time for Android (per config) is comparable to the build + test time for other configurations. Not bad at all. I imagine we'd be in pretty good shape with just the ARM configuration that builds the library + Java bindings?

File: workspaces/Android-NDK/jni/Android.mk [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+LOCAL_MODULE    := assimp_static
+ASSIMP_SRC_DIR = code
+FILE_LIST := $(wildcard $(LOCAL_PATH)/$(ASSIMP_SRC_DIR)/*.cpp)
+
+IGNORE_LIST := $(wildcard $(LOCAL_PATH)/$(ASSIMP_SRC_DIR)/C4D*.cpp)
+EXCLUDES := $(IGNORE_LIST:$(LOCAL_PATH)/%=%)
+ALL_SRC_FILES := $(FILE_LIST:$(LOCAL_PATH)/%=%)
+LOCAL_SRC_FILES := $(filter-out $(EXCLUDES), $(ALL_SRC_FILES))
+
+LOCAL_SRC_FILES += contrib/clipper/clipper.cpp \
```
Comment (acgessler) [inline workspaces/Android-NDK/jni/Android.mk]: By any chance, can we wildcard those as well?

File: workspaces/Android-NDK/jni/Android.mk [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+	contrib/unzip/ioapi.c \
+	contrib/unzip/unzip.c \
+	contrib/irrXML/irrXML.cpp \
+	contrib/poly2tri/poly2tri/sweep/advancing_front.cc \
+	contrib/poly2tri/poly2tri/sweep/sweep_context.cc \
+	contrib/poly2tri/poly2tri/sweep/sweep.cc \
+	contrib/poly2tri/poly2tri/sweep/cdt.cc \
+	contrib/poly2tri/poly2tri/common/shapes.cc
+
+LOCAL_SRC_FILES += contrib/openddlparser/code/OpenDDLParser.cpp \
```
Comment (acgessler) [inline workspaces/Android-NDK/jni/Android.mk]: Dito.

**Comment by acgessler** [comment]
Thanks, looks great. Why do we end up with two ANDROID builds in Travis though? What's the difference?

**Comment by arris69** [comment]
two builds came from compiler matrix? clang and cpp thats why it ends also in 8 windows and 8 linux builds...

**Comment by acgessler** [comment]
I see, sure. Thanks for updating!

**Pull request merged**